src/components/Checkbox/Checkbox.tsx

```diff
@@ -112,6 +115,21 @@ const CheckboxComponent: ForwardRefRenderFunction<HTMLInputElement, CheckboxProp
         inputRef,
     );
 
+    const getCheckboxSize = () => {
```

Comment:
Here instead of creating a function with a switch you could use a variable with Record ->
```
const sizeClassesMap: Record<CheckboxSize, string> = {
    [CheckboxSize.Default]: 'tw-h-4 tw-w-4',
    [CheckboxSize.Large]: 'tw-h-5 tw-w-5',
    [CheckboxSize.XLarge]: 'tw-h-8 tw-w-8',
}
```